interfaces/mount/spec.go

```diff
@@ -158,33 +159,115 @@ func (spec *Specification) MountEntries() []osutil.MountEntry {
 	result = append(result, spec.overname...)
 	result = append(result, spec.layout...)
 	result = append(result, spec.general...)
-	unclashMountEntries(result)
-	return result
+	return unclashMountEntries(result)
 }
 
 // UserMountEntries returns a copy of the added user mount entries.
 func (spec *Specification) UserMountEntries() []osutil.MountEntry {
 	result := make([]osutil.MountEntry, len(spec.user))
 	copy(result, spec.user)
-	unclashMountEntries(result)
-	return result
+	return unclashMountEntries(result)
+}
+
+// Assuming that two mount entries have the same source, target and type, this
+// function computes the mount options that should be used when performing the
+// mount, so that the most permissive options are kept.
+// The following flags are considered (of course the operation is commutative):
+//   - "ro" + "rw" = "rw"
+//   - "bind" + "rbind" = "rbind
+func mergeOptions(options ...[]string) []string {
+	mergedOptions := make([]string, 0, len(options[0]))
+	foundWritableEntry := false
+	foundRBindEntry := false
+	firstEntryIsBindMount := false
+	for i, opts := range options {
+		isReadOnly := false
+		isRBind := false
+		for _, o := range opts {
+			switch o {
+			case "ro":
+				isReadOnly = true
+			case "rbind":
+				isRBind = true
+				fallthrough
+			case "bind":
+				if i == 0 {
```

Comment:
I'm missing something, I'm not sure why we consider only bind from the first options set, also the function comment doesn't say anything about that, I would assume the logic is the same for ro+rw

Reply:
I added a comment, let me know if it makes sense.

Reply:
I see, not related to this PR but we don't have a check that bind MountEntries added to spec don't set Type? and that if Type is set bind/rbind are not used? I suppose usage tests and reviews cover that in a more process way.

Anyway I just wonder if the name should just isBindMount then, also the code would also work without the if as well afaict